rmw_zenoh_cpp/src/detail/rmw_client_data.cpp

```diff
@@ -244,46 +176,27 @@ ClientData::ClientData(
 }
 
 ///=============================================================================
-bool ClientData::init(std::shared_ptr<ZenohSession> session)
+bool ClientData::init(const std::shared_ptr<zenoh::Session> session)
```

Comment:
This is an existing problem, but we may as well fix it; since we are now passing the session into the `ClientData` constructor, we don't need to pass it here.  We already have the session in `sess_`.

```suggestion
bool ClientData::init()
```

Reply:
(further, I think that given what we have changed elsewhere we might be able to get rid of this `init` method completely, but I'm going to wait until we do the lambda captures before asking to remove this)